File: wup1/Python/sum_double.py
'''
    sum_double 
    
    Given two int values, return their sum. Unless the two values are the same, 
    then return double their sum.
'''

def sum_double(a, b):
    if a == b:
        return (a + b) * 2
    else:
        return a + b
        
def tests():
    assert (sum_double(1, 2) == 3)
    assert (sum_double(3, 2) == 5)
    assert (sum_double(2, 2) == 8)
    assert (sum_double(-1, 0) == -1)
    assert (sum_double(3, 3) == 12)
    assert (sum_double(0, 0) == 0)
    assert (sum_double(0, 1) == 1)
    assert (sum_double(3, 4) == 7)

    print("All tests passed")
    
tests()

File: wup1/JS/SumDouble.js
/* sumDouble 
    Given two int values, return their sum. Unless the two values are the same, 
    then return double their sum. */

function sumDouble(a, b){
  if (a === b)
    return (a+b) * 2;
  else
    return a+b;
}

const assert = require('assert');
assert.equal(sumDouble(1, 2), 3);
assert.equal(sumDouble(3, 2), 5);
assert.equal(sumDouble(2, 2), 8);
assert.equal(sumDouble(-1, 0), -1);
assert.equal(sumDouble(3, 3), 12);
assert.equal(sumDouble(0, 0), 0);
assert.equal(sumDouble(0, 1), 1);
assert.equal(sumDouble(3, 4), 7);

console.log("All tests passed");